src/WebJobs.Script/Diagnostics/FileLogger.cs

```diff
@@ -107,7 +108,14 @@ private static TraceLevel GetTraceLevel(LogLevel logLevel)
 
         private static bool IsFromTraceWriter(IEnumerable<KeyValuePair<string, object>> properties)
         {
-            return properties.Any(kvp => kvp.Key == ScriptConstants.TracePropertyIsUserTraceKey);
+            if (properties == null)
+            {
+                return false;
+            }
+            else
+            {
+                return properties.Any(kvp => kvp.Key == ScriptConstants.TracePropertyIsUserTraceKey);
```

Comment:
I know this is previously existing code, but it would be nice to change this to `string.Equals` with a proper comparison passed in